### ISSUE TITLE
Feature Request: Server Port Filtering in SteamUser

### ISSUE THREAD

**Issue by koftrciali** [DoctorMcKay/node-steam-user#488]
### Problem Statement
Some proxy providers, such as [Thunderproxies](https://thunderproxy.com/), restrict access to all ports except 443 and 80. This limitation makes it challenging to establish connections through other ports.

### Proposed Solution
Introduce a serverFilter function as an option when creating a SteamUser instance. This function would allow users to filter the server list based on specific criteria, such as port availability.

```js
const client = new SteamUser({
    serverFilter: (serverList) => {
        return serverList.filter(server => server.port === 443 || server.port === 80);
    }
});
```
### Alternatives Considered
None at this time.

### Additional Context
Implementing this feature would simplify connecting to services that restrict port usage, ensuring compatibility with a broader range of proxy providers.

**Comment by DoctorMcKay** [comment]
Sounds like [webCompatibilityMode](https://github.com/DoctorMcKay/node-steam-user?tab=readme-ov-file#webcompatibilitymode) does what you want?

**Comment by Sadzurami** [comment]
> Sounds like [webCompatibilityMode](https://github.com/DoctorMcKay/node-steam-user?tab=readme-ov-file#webcompatibilitymode) does what you want?

looks this feature not working as expected
after retrieving CMServers from api, servers list not filters by 443 port
so, even if `webCompatibilityMode` is enabled, we still can get websocket with 27*** port

**Comment by DoctorMcKay** [comment]
> > Sounds like [webCompatibilityMode](https://github.com/DoctorMcKay/node-steam-user?tab=readme-ov-file#webcompatibilitymode) does what you want?
> 
> looks this feature not works as expected after retrieving CMServers from api, servers list not filters by 443 port so, even if `webCompatibilityMode` is enabled, we still can get websocket with 27*** port

Fixed in [5.1.0](https://github.com/DoctorMcKay/node-steam-user/releases/tag/v5.1.0)
